pkg/verification/service/verification_service.go

```diff
@@ -228,12 +228,27 @@ func (s *ServiceImpl) VerifyPhoneCode(ctx *gin.Context, userID, username, code s
 		return crterrors.NewInternalError(lookupErr, fmt.Sprintf("error retrieving usersignup: %s", userID))
 	}
 
-	// require manual approval if captcha score below automatic verification threshold
-	captchaScore, found := signup.Annotations[toolchainv1alpha1.UserSignupCaptchaScoreAnnotationKey]
-	fscore, parseErr := strconv.ParseFloat(captchaScore, 32)
-	if found && parseErr == nil && fscore < 0.6 {
-		log.Error(ctx, errors.New("captcha score too low"), "automatic verification disabled, manual approval required for user")
-		return crterrors.NewForbiddenError("verification failed", "verification is not available at this time")
+	// check if it's a reactivation
+	if activationCounterString, foundActivationCounter := signup.Annotations[toolchainv1alpha1.UserSignupActivationCounterAnnotationKey]; foundActivationCounter && cfg.Verification().CaptchaAllowLowScoreReactivation() {
+		activationCounter, err := strconv.Atoi(activationCounterString)
+		if err != nil {
+			log.Error(ctx, err, "activation counter is not an integer value, checking required captcha score")
+			// require manual approval if captcha score below automatic verification threshold
+			if err = checkRequiredCaptchaScore(ctx, signup, cfg); err != nil {
+				return err
+			}
+		} else if activationCounter == 1 {
+			// check required captcha score if it's not a reactivation
+			if err = checkRequiredCaptchaScore(ctx, signup, cfg); err != nil {
+				return err
+			}
+		}
+	} else {
+		// when allowLowScoreReactivation is not enabled or no activation counter found
+		// require manual approval if captcha score below automatic verification threshold for all users
+		if err := checkRequiredCaptchaScore(ctx, signup, cfg); err != nil {
+			return err
+		}
```

Comment:
I may be missing something but it looks like all cases lead to calling `checkRequiredCaptchaScore(ctx, signup, cfg)`. If CaptchaAllowLowScoreReactivation is true and it's a reactivation then doesn't that mean we don't need to do additional (phone) verification and we can just proceed to activate the user?

So maybe we can simplify the logic by removing the CaptchaAllowLowScoreReactivation check from here and move it to the IsPhoneVerificationRequired function in https://github.com/codeready-toolchain/registration-service/blob/1602277a54b638f1056f52ef77757f41cc61d2d5/pkg/signup/service/signup_service.go#L227C2-L232 so that we skip phone verification at that point instead and keep the logic simple here?

Reply:
> I may be missing something but it looks like all cases lead to calling checkRequiredCaptchaScore(ctx, signup, cfg).

the case in which is `activationCounter > 1` and `CaptchaAllowLowScoreReactivation == true` doesn't lead to `checkRequiredCaptchaScore` . Maybe I can shuffle the if statements a bit to make this more clear 🤔

Reply:
>Not sure, we could move it there but we need to pass the activation counter from the user. Do you think that would make it simpler ?

You're right, that part of the code doesn't reference the UserSignup so I guess it's more complicated. My bad, disregard the suggestion.